library/x509_create.c

```diff
@@ -218,6 +218,8 @@ static int parse_attribute_value_hex_der_encoded(const char *s,
                                                  size_t *data_len,
                                                  int *tag)
 {
+    unsigned char *p;
```

Comment:
I mostly agree, but leaving it uninitialised was a deliberate choice here, in that if we accidentally modified the code such that it tried to use it, we would get a compiler warning, whereas if we set it to NULL, we won't.

Reply:
I used to think that way, but I've found that compilers often don't signal potentially uninitialized variables when they don't know for sure whether the variable is initialized on all code paths where it's used. It happens from time to time that we merge code that the CI accepts, then Coverity signals a code path where a variable isn't initialized. With a pointer, this can be devastating.

Reply:
> Blocker: Please don't leave a pointer variable uninitialized. That creates a significant risk that at some point, we'll refactor the code and have some cleanup that accesses the pointer at a point where it's uninitialized in rare circumstances.

Since this would have been a blocker, can you add something to the coding standards to explicitly state this?

Reply:
I consider what you did in https://github.com/Mbed-TLS/mbedtls-docs/pull/122 sufficient for now.